packages/types/src/signer.ts

```diff
@@ -0,0 +1,24 @@
+import { Account } from "./account";
+import {
+  IbcTransferProps,
+  InitAccountProps,
+  SignedTx,
+  TransferProps,
+  TxProps,
+} from "./tx";
+
+export interface Signer<T = Account> {
+  accounts: () => Promise<T[] | undefined>;
+  signTx(
+    signer: string,
+    txProps: TxProps,
+    encodedTx: string,
+    txData: string
+  ): Promise<SignedTx | undefined>;
```

Comment:
Is `encodedTx: string` intentional here? When I do a global search for `encodedTx` this is the only hit I get. Due to the structural type system we are allowed to omit this in the implementing class and we are omitting it in `class Signer implements ISigner` in `namada-interface/apps/extension/src/provider/Signer.ts`.
```suggestion
  signTx(
    signer: string,
    txProps: TxProps,
    txData: string
  ): Promise<SignedTx | undefined>;
```